front/lib/api/assistant/actions/database_query.ts

```diff
@@ -0,0 +1,250 @@
+      created: Date.now(),
+      configurationId: configuration.sId,
+      messageId: agentMessage.sId,
+      error: {
+        code: "database_query_parameters_generation_error",
+        message: "Cannot access the database linked to this action.",
+      },
+    };
+  }
+
+  // Generating inputs
+  const inputRes = await generateDatabaseQueryAppParams(
+    auth,
+    configuration,
+    conversation,
+    userMessage
+  );
+  if (inputRes.isErr()) {
+    yield {
+      type: "database_query_run_error",
+      created: Date.now(),
+      configurationId: configuration.sId,
+      messageId: agentMessage.sId,
+      error: {
+        code: "database_query_parameters_generation_error",
+        message: `Error generating parameters for database_query: ${inputRes.error.message}`,
+      },
+    };
+    return;
+  }
+  const input = inputRes.value;
+
+  // Generating configuration
+  const config = cloneBaseConfig(
+    DustProdActionRegistry["assistant-v2-query-database"].config
+  );
+  const database = {
+    workspace_id: c.dataSourceWorkspaceId,
+    data_source_id: c.dataSourceId,
+    database_id: c.databaseId,
+  };
+  config.DATABASE_SCHEMA = {
+    type: "database_schema",
+    database,
+  };
+  config.DATABASE = {
+    type: "database",
+    database,
+  };
+
+  // Running the app
+  const res = await runActionStreamed(
+    auth,
+    "assistant-v2-query-database",
+    config,
+    [{ question: input.question }]
+  );
+
+  if (res.isErr()) {
+    yield {
+      type: "database_query_run_error",
+      created: Date.now(),
+      configurationId: configuration.sId,
+      messageId: agentMessage.sId,
+      error: {
+        code: "dust_app_run_error",
+        message: `Error running DatabaseQuery app: ${res.error.message}`,
+      },
+    };
+    return;
+  }
+
+  const { eventStream } = res.value;
+  for await (const event of eventStream) {
+    if (event.type === "block_execution") {
+      const e = event.content.execution[0][0];
+      if (event.content.block_name === "OUTPUT" && e.value) {
+        const output = JSON.parse(e.value as string);
+
+        const action = await AgentDatabaseQueryAction.create({
+          dataSourceWorkspaceId: c.dataSourceWorkspaceId,
+          dataSourceId: c.dataSourceId,
+          databaseId: c.databaseId,
+          databaseQueryConfigurationId: configuration.sId,
+          params: input,
+          output,
+        });
+        yield {
+          type: "database_query_run_success",
+          created: Date.now(),
+          configurationId: configuration.sId,
+          messageId: agentMessage.sId,
+          action: {
+            id: action.id,
+            type: "database_query_action",
+            dataSourceWorkspaceId: action.dataSourceWorkspaceId,
+            dataSourceId: action.dataSourceId,
+            databaseId: action.databaseId,
+            output: action.output,
+          },
+        };
+        return;
+      }
+    }
+  }
+
+  yield {
```

Comment:
I think it's a bit distrurbing to conclude with an error, let's emit the success event here, capturing the data in the event loop, as is the case in other use of these event streams.